graphql_compiler/cost_estimation/statistics.py

```diff
@@ -1,10 +1,28 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
 import datetime
+import math
+from typing import Any, Dict, List, Optional, Tuple
 
 import six
 
 
+@dataclass
+class SamplingSummary:
+    """Results extracted from sampling a random rowset that are relevant for statistics."""
+
+    # The class this summary describes
+    class_name: str
+
+    # Mapping field_name -> field_value -> observed_count for all field names and observed
```

Comment:
```suggestion
    # Mapping field_name -> (field_value, observed_count) for all field names and observed
```

Reply:
That would be `Dict[str, Tuple[Any, int]]`. It's a different thing. I can add parens if it's not implied: `field_name -> (field_value -> observed_count)`

Reply:
^ This is the only legal interpretation since the alternative interpretation (field_name -> field_value) -> observed_count would have a dict as a key which is not allowed because dicts are mutable (although you're free to add parens if you feel it helps readability! I think it probably does)

Reply:
I'm used to `->` being right associative from Haskell/Idris, but there's no reason to bring those norms into a python codebase, so I'll add parens for clarity.